scripts/windows/utils_functions.ps1

```diff
@@ -0,0 +1,186 @@
+   #!/bin/bash
+
+#################################################################################################################################
+#################################################### Utils Windows Functions ####################################################
+#################################################################################################################################
+
+# Prints error message in red
+# Input:
+#   message - Message text
+# Output:
+#   The message
+function Write-Error ([string]$message) {
+    Write-Log "ERROR" "$message"
+    Write-Host "$message" -ForegroundColor Red
+}
+
+# Prints warning message in yellow
+# Input:
+#   message - Message text
+# Output:
+#   The message
+function Write-Warning ([string]$message) {
+    Write-Log "WARN" "$message"
+    Write-Host "$message" -ForegroundColor Yellow
+}
+
+# Writes log into Logz.io agent log file
+# Input:
+#   logLevel - The level of the log (INFO/ERROR/WARN)
+#   log - Log text
+function Write-Log ([string]$logLevel, [string]$log) {
+    Write-Output "[$logLevel] [$(Get-Date -Format "yyyy-MM-dd HH:mm:ss")] $log" >> $logFile
+}
+
+# Writes command into run file in Logz.io temp directory
+# Input:
+#   command - The command to write into the file
+function Write-Run ([string]$command) {
+    Write-Output "$command" >> $runFile
+}
+
+# Deletes the temp directory
+function Remove-TempDir {
+    if (Test-Path $logzioTempDir) {
+        Remove-Item -Path $logzioTempDir -Recurse
+    }
+}
+
+# Gets task error file error message
+# Output:
+#   The task error message
+function Get-TaskError {
+    $local:err = Get-Content $taskErrorFile -First 1
+    if ([string]::IsNullOrEmpty($err)) {
+        return
+    }
+
+    $err = $err.Replace("`"", "'")
+    Write-Output "$err"
+}
+
+# Finds the requested parameter in params 
+# Inputs: 
+#   params - The parameters in the application json
+#   requestedName - The parameter name to find
+# Output:
+#   The requested parameter if requestedName was found, empty otherwise.
+function Find-Param ([string]$params, [string]$requestedName) {
+    $local:paramsList = Write-Output $params | jq -c '.'
+    $local:requestedParam = ""
+
+    foreach ($param in $paramsList) {
+        $local:name = Write-Output "$param" | jq -r '.name'
+        if ([string]::IsNullOrEmpty($name) -or $name.Equals("null")) {
+            continue
+        }
+        
+        if ($name.Equals($requestedName)) {
+            $requestedParam = "$param"
+        }
+    }
+
+    Write-Output "$requestedParam"
+}
+
+# Installs Chocolatey
+# Error:
+#   Exit Code 1
+function Install-Chocolatey {
```

Comment:
why we need Chocolatey

Reply:
if jq/helm and chocolatey are not installed, we will install chocolatey to install them.